From 6dcb32201ac963082a32fb62ef1235f7fe23ba4a Mon Sep 17 00:00:00 2001
From: "Shawn.Luo" <strugglelg@gmail.com>
Date: Thu, 22 Nov 2018 10:05:26 +0800
Subject: [Lwip] init name first


diff --git a/lwip-1.4.0/src/core/netif.c b/lwip-1.4.0/src/core/netif.c
index fa205d5..6e3fc00 100644
--- a/lwip-1.4.0/src/core/netif.c
+++ b/lwip-1.4.0/src/core/netif.c
@@ -133,6 +133,8 @@ netif_init(void)
  *
  * @return netif, or NULL if failed.
  */
+ #include "netif/ethernetif.h"
+
 struct netif *
 netif_add(struct netif *netif, ip_addr_t *ipaddr, ip_addr_t *netmask,
   ip_addr_t *gw, void *state, netif_init_fn init, netif_input_fn input)
@@ -141,6 +143,10 @@ netif_add(struct netif *netif, ip_addr_t *ipaddr, ip_addr_t *netmask,
 
   LWIP_ASSERT("No init function given", init != NULL);
 
+  /* init name;	Copy from  lwip-1.4.0/src/netif/ethernetif.c */
+  netif->name[0] = IFNAME0;
+  netif->name[1] = IFNAME1;
+
   /* reset new interface configuration state */
   ip_addr_set_zero(&netif->ip_addr);
   ip_addr_set_zero(&netif->netmask);
@@ -222,8 +228,13 @@ void
 netif_set_addr(struct netif *netif, ip_addr_t *ipaddr, ip_addr_t *netmask,
     ip_addr_t *gw)
 {
+  printf("\x1b[32m##<<======= netif_set_ipaddr() =======>>##\x1b[0m\n");
   netif_set_ipaddr(netif, ipaddr);
+
+  printf("\x1b[32m##<<======= netif_set_netmask() =======>>##\x1b[0m\n");
   netif_set_netmask(netif, netmask);
+
+  printf("\x1b[32m##<<======= netif_set_gw() =======>>##\x1b[0m\n");
   netif_set_gw(netif, gw);
 }
 
